compiler/qsc/src/codegen/tests.rs

```diff
@@ -525,6 +525,145 @@ mod adaptive_profile {
         "#]].assert_eq(&qir);
     }
 
+    #[test]
+    fn custom_measurement_generates_correct_qir() {
+        let source = "namespace Test {
+            operation Main() : Result {
+                use q = Qubit();
+                H(q);
+                __quantum__qis__mx__body(q)
+            }
+
+            @Measurement()
+            operation __quantum__qis__mx__body(target: Qubit) : Result {
+                body intrinsic;
+            }
+        }";
+        let sources = SourceMap::new([("test.qs".into(), source.into())], None);
+        let language_features = LanguageFeatures::default();
+        let capabilities = TargetCapabilityFlags::Adaptive;
+
+        let (std_id, store) = crate::compile::package_store_with_stdlib(capabilities);
+        let qir = get_qir(
+            sources,
+            language_features,
+            capabilities,
+            store,
+            &[(std_id, None)],
+        )
+        .expect("Failed to generate QIR");
```

Comment:
nitpick: I mentioned this on another PR, but I think it's worth keeping in mind some guidance from the style recommendation in the Rust stdlib (see [Common Message Styles](https://doc.rust-lang.org/std/error/index.html#common-message-styles)) that "expect as precondition" is preferable. It's not critical, but I do think that style makes the readability of the code and the diagnosibility of the tests a little nicer. Food for thought, not worth blocking the PR over.